examples/training/contrastive/imagenet/simclr_training.py

```diff
@@ -0,0 +1,192 @@
+)
+flags.DEFINE_string("imagenet_path", None, "Directory from which to load Imagenet.")
+flags.DEFINE_string(
+    "backup_path", None, "Directory which will be used for training backups."
+)
+flags.DEFINE_string(
+    "weights_path", None, "Directory which will be used to store weight checkpoints."
+)
+flags.DEFINE_string(
+    "tensorboard_path", None, "Directory which will be used to store tensorboard logs."
+)
+flags.DEFINE_integer("batch_size", 256, "Batch size for training and evaluation.")
+flags.DEFINE_boolean(
+    "use_xla", True, "Whether or not to use XLA (jit_compile) for training."
+)
+flags.DEFINE_float(
+    "initial_learning_rate",
+    0.1,
+    "Initial learning rate which will reduce on plateau.",
+)
+flags.DEFINE_boolean(
+    "include_probe",
+    True,
+    "Whether to include probing during training.",
+)
+
+
+FLAGS = flags.FLAGS
+FLAGS(sys.argv)
+
+if FLAGS.model_name not in models.__dict__:
+    raise ValueError(f"Invalid model name: {FLAGS.model_name}")
+
+CLASSES = 1000
+IMAGE_SIZE = (224, 224)
+EPOCHS = 250
+
+
+def parse_imagenet_example(example):
+    # Read example
+    image_key = "image/encoded"
+    label_key = "image/class/label"
+    keys_to_features = {
+        image_key: tf.io.FixedLenFeature((), tf.string, ""),
+        label_key: tf.io.FixedLenFeature([], tf.int64, -1),
+    }
+    parsed = tf.io.parse_single_example(example, keys_to_features)
+
+    # Decode and resize image
+    image_bytes = tf.reshape(parsed[image_key], shape=[])
+    image = tf.io.decode_jpeg(image_bytes, channels=3)
+    image = layers.Resizing(
+        width=IMAGE_SIZE[0], height=IMAGE_SIZE[1], crop_to_aspect_ratio=True
+    )(image)
+
+    # Decode label
+    label = tf.cast(tf.reshape(parsed[label_key], shape=()), dtype=tf.int32) - 1
+    label = tf.one_hot(label, CLASSES)
+    if FLAGS.include_probe:
+        return image, label
+    else:
+        return image
+
+
+def load_imagenet_dataset():
+    train_filenames = [
+        f"{FLAGS.imagenet_path}/train-{i:05d}-of-01024" for i in range(0, 1024)
+    ]
+    validation_filenames = [
+        f"{FLAGS.imagenet_path}/validation-{i:05d}-of-00128" for i in range(0, 128)
+    ]
+
+    train_dataset = tf.data.TFRecordDataset(filenames=train_filenames)
+    validation_dataset = tf.data.TFRecordDataset(filenames=validation_filenames)
+
+    train_dataset = train_dataset.map(
+        parse_imagenet_example,
+        num_parallel_calls=tf.data.AUTOTUNE,
+    ).shuffle(2000, reshuffle_each_iteration=True)
+    validation_dataset = validation_dataset.map(
+        parse_imagenet_example,
+        num_parallel_calls=tf.data.AUTOTUNE,
+    ).shuffle(2000, reshuffle_each_iteration=True)
+
+    train_dataset = train_dataset.batch(FLAGS.batch_size).prefetch(tf.data.AUTOTUNE)
+    validation_dataset = validation_dataset.batch(FLAGS.batch_size).prefetch(
+        tf.data.AUTOTUNE
+    )
+
+    return train_dataset, validation_dataset
+
+
+train_ds, test_ds = load_imagenet_dataset()
```

Comment:
lets add an entry to `keras_cv.datasets.imagenet` and document how to prepare the tfrecords.

It would be great to streamline imagenet loading for users, perhaps even provide a script to generate them.

Reply:
Done in #735.

Reply:
Once 735 is merged, I'll update this script to use `keras_cv.datasets.imagenet` to load the data.